### PR TITLE
feat: support `slowTestThreshold` option

### PR DESCRIPTION
**Pull request by mrazauskas**
Just could not stop myself – added support for `slowTestThreshold` option.

Had to rewrite the logic, but that is even better after types were added. I mean, in some files there are more types than code. After refactor all is much cleaner.

### CODE REVIEW

File: src/run.js [hunk 1]
```diff
@@ -1,38 +1,81 @@
...
+ * @param {number} duration
+ * @param {string=} errorMessage
+ * @returns {TestCaseResult}
+ */
+function toJestResult(status, duration, errorMessage) {
+  return {
+    ancestorTitles: [],
+    failureDetails: [],
+    failureMessages: errorMessage ? [errorMessage] : [],
+    fullName: TEST_TITLE,
```
Comment (mrazauskas) [inline src/run.js:19]: Currently `fullName` is set to absolute path of the test file. Not sure if this is a breaking change, or just a fix?

Comment (SimenB) [reply src/run.js:19]: seems fine - we expose the filename via `testFilePath`

**Review by SimenB — APPROVED** [review]
love it :+1:

**Comment by SimenB** [comment]
https://github.com/jest-community/jest-runner-tsd/releases/tag/v3.1.0

**Pull request merged**